vars/uploadMinSnapshotsToS3.groovy

```diff
@@ -1,15 +1,42 @@
 void call(Map args = [:]) {
+
     def lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+    List<Closure> fileActions = args.fileActions ?: []
     String manifest = args.manifest ?: "manifests/${INPUT_MANIFEST}"
+
     def inputManifest = lib.jenkins.InputManifest.new(readYaml(file: manifest))
     echo("Retreving build manifest from: $WORKSPACE/builds/${inputManifest.build.getFilename()}/manifest.yml")
-    def buildManifest = lib.jenkins.BuildManifest.new(readYaml(file: "$WORKSPACE/builds/${inputManifest.build.getFilename()}/manifest.yml"))
+
+    productName = inputManifest.build.getFilename()
+    def buildManifest = lib.jenkins.BuildManifest.new(readYaml(file: "$WORKSPACE/builds/${productName}/manifest.yml"))
     version = buildManifest.build.version
     architecture = buildManifest.build.architecture
     platform = buildManifest.build.platform
-    minArtifactSourceName = "opensearch-min-${version}-${platform}-${architecture}.tar.gz"
-    minArtifactName = "opensearch-min-${version}-${platform}-${architecture}-latest.tar.gz"
+    id = buildManifest.build.id
+
+    // Setup src & dst variables for artifacts
+    srcDir = "${WORKSPACE}/builds/${productName}/dist"
+    dstDir = "snapshots/core/${productName}/${version}"
+    baseName = "${productName}-min-${version}-${platform}-${architecture}"
+    tarSrcName = "${baseName}.tar.gz"
+    tarDstNameId = "${baseName}-${id}.tar.gz"
+    tarDstNameLatest = "${baseName}-latest.tar.gz"
+    shaSrcName = "${tarSrcName}.sha512"
+    shaDstNameId = "${tarDstNameId}.sha512"
+    shaDstNameLatest = "${tarDstNameLatest}.sha512"
+
+    // Create checksums
+    echo("Create .sha512 for Min Snapshots Artifacts")
+    argsMap = [:]
+    argsMap['artifactPath'] = srcDir
+    for (Closure action : fileActions) { // running createSha512Checksums()
```

Comment:
Nit: Does it run `createSha512Checksums`? If so, then a) hard-code that value here b) remove it from `args` c) default `fileArgs` above to it. 

At least either remove the comment. or change the comment to something like (`e.g. createSha512Checksums()`).

Reply:
It is written that way in case new actions such as createSignature() or more can be called as well.
The same logic has been used in promotion lib.
If you find it to be too complicated I can change it, but other than that I think it is ok.